tests/unit/data-stores/test-class-wcs-related-order-store-cached-cpt.php

```diff
@@ -241,6 +241,70 @@ public function test_delete_relations( $relation_type ) {
 		}
 	}
 
+	/**
+	 * Make sure a related order is removed from the cached relationship array after deletion.
+	 *
+	 * @dataProvider provider_relation_type
+	 */
+	public function test_cache_updates_after_related_object_is_deleted_one_to_many( $relation_type ) {
+		$subscription       = WCS_Helper_Subscription::create_subscription();
+		$order_to_delete    = WCS_Helper_Subscription::create_order();
+		$order_id_to_delete = $order_to_delete->get_id();
+		$order_to_keep      = WCS_Helper_Subscription::create_order();
+		$order_id_to_keep   = $order_to_keep->get_id();
+
+		self::$cache_store->add_relation( $order_to_delete, $subscription, $relation_type );
+		self::$cache_store->add_relation( $order_to_keep, $subscription, $relation_type );
+
+		// Verify that the related orders are correct and cached.
+		$related_order_ids = self::$cache_store->get_related_order_ids( $subscription, $relation_type );
+		$this->assertContains( $order_id_to_keep, $related_order_ids );
+		$this->assertContains( $order_id_to_delete, $related_order_ids );
+
+		$order_to_delete->delete( true );
+
+		// Verify that the deleted order was removed from the cached relationship
+		$related_order_ids = self::$cache_store->get_related_order_ids( $subscription, $relation_type );
+		$this->assertContains( $order_id_to_keep, $related_order_ids );
+		$this->assertNotContains( $order_id_to_delete, $related_order_ids );
+	}
+
+	/**
+	 * Make sure a related order is removed from the cached relationship array after deletion when the order is related
+	 * to more than one Subscription.
+	 *
+	 * @dataProvider provider_relation_type
+	 */
+	public function test_cache_updates_after_related_object_is_deleted_many_to_many( $relation_type ) {
```

Comment:
This second test currently fails with HPOS enabled because of the following.

https://github.com/Automattic/woocommerce-subscriptions-core/blob/39fc2b3394f207154fd29bbf78b7b3ad41b889b4/includes/class-wcs-object-data-cache-manager.php#L157-L158

@mattallan Do you know if it is a valid case that an order can be related to more than one subscription and end up with multiple MetaData entries for a relationship?  If not, we can remove the second test.  It is possible, we may need to restructure the way object_changes are stored to allow for more than one meta value to change per object/key pair.

Reply:
Hey @prettyboymp, thanks for this PR.

> Do you know if it is a valid case that an order can be related to more than one subscription and end up with multiple MetaData entries for a relationship?

No there isn't a case to my knowledge where we have multiple metadata entries for a one-order-to-many-subscription relationship.

It is possible to have multiple subscriptions related to a single order i.e. a parent orders can have multiple subscriptions but we don't store these under multiple metadata entries. For these cases, the `parent_order_id` for each order would have the same subscription ID.

Reply:
> No there isn't a case to my knowledge where we have multiple metadata entries for a one-order-to-many-subscription relationship.

Okay.  Then I'll remove this second test and then PR should be good for final review.